src/daal4py.cpp

```diff
@@ -427,24 +491,22 @@ daal::data_management::NumericTablePtr make_nt(PyObject * obj)
                 if(PyErr_Occurred()) {PyErr_Print(); throw std::runtime_error("Python Error");}
 
                 if(np_indcs && np_roffs && np_vals && nr && nc) {
-                    // for now, increment indcs by 1
                     size_t * c_indcs = (size_t*)array_data(np_indcs);
                     size_t n = array_size(np_indcs, 0);
-                    for(size_t i=0; i<n; ++i) c_indcs[i] += 1;
-
+                    size_t * c_indcs_one_based = (size_t*)malloc(n * sizeof(size_t));
+                    for(size_t i=0; i<n; ++i) c_indcs_one_based[i] = c_indcs[i] + 1;
                     size_t * c_roffs = (size_t*)array_data(np_roffs);
                     n = array_size(np_roffs, 0);
-                    for(size_t i=0; i<n; ++i) c_roffs[i] += 1;
-
+                    size_t * c_roffs_one_based = (size_t*)malloc(n * sizeof(size_t));
+                    for(size_t i=0; i<n; ++i) c_roffs_one_based[i] = c_roffs[i] + 1;
                     size_t c_nc = (size_t)PyInt_AsSsize_t(nc);
                     if(PyErr_Occurred()) {PyErr_Print(); throw std::runtime_error("Python Error");}
                     size_t c_nr = (size_t)PyInt_AsSsize_t(nr);
                     if(PyErr_Occurred()) {PyErr_Print(); throw std::runtime_error("Python Error");}
-
 #define MKCSR_(_T)                                                      \
                     ret = daal::data_management::CSRNumericTable::create(daal::services::SharedPtr<_T>((_T*)array_data(np_vals), NumpyDeleter((PyArrayObject*)np_vals)), \
-                                                                         daal::services::SharedPtr<size_t>(c_indcs, NumpyDeleter((PyArrayObject*)np_indcs)), \
-                                                                         daal::services::SharedPtr<size_t>(c_roffs, NumpyDeleter((PyArrayObject*)np_roffs)), \
+                                                                         daal::services::SharedPtr<size_t>(c_indcs_one_based, VoidDeleter()), \
```

Comment:
Use [service deleter](https://github.com/oneapi-src/oneDAL/blob/master/cpp/daal/include/services/daal_shared_ptr.h#L82) from DAAL:
```suggestion
                                                                         daal::services::SharedPtr<size_t>(c_indcs_one_based, daal::services::ServiceDeleter()), \
```